R/pkg/R/DataFrame.R

```diff
@@ -1324,12 +1312,16 @@ setMethod("selectExpr",
 #' path <- "path/to/file.json"
 #' df <- jsonFile(sqlContext, path)
 #' newDF <- withColumn(df, "newCol", df$col1 * 5)
+#' # Replace an existing column
+#' newDF2 <- withColumn(newDF, "newCol", newDF$col1)
```

Comment:
I know the reason. When the withColumn was implemented in SparkR, the withColumn() in Scala support just adding columns, without support for replacing existing columns. But later, withColumn() in Scala was enhanced to support replacing existing columns, see https://github.com/apache/spark/pull/5541.  However, withColumn in SparkR have not been synced with Scala until this PR:)